zio-config/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -2,12 +2,76 @@ package zio.config
 
 import zio.IO
 
-trait ConfigSource {
+trait ConfigSource { self =>
   val configService: ConfigSource.Service
+
+  final def orElse(that: ConfigSource): ConfigSource =
+    new ConfigSource {
+      val configService: ConfigSource.Service =
+        new ConfigSource.Service {
+          val sourceDescription: String = {
+            val selfDesc = self.configService.sourceDescription
+            val thatDesc = that.configService.sourceDescription
+
+            s"${selfDesc}: with fallback source: ${thatDesc} (on all errors)"
+          }
+
+          def getConfigValue(path: String): IO[ReadError, ConfigSource.Value] =
+            self.configService
+              .getConfigValue(path)
+              .orElse(
+                that.configService
+                  .getConfigValue(path)
+              )
+        }
+    }
+
+  final def <>(that: ConfigSource): ConfigSource =
+    orElse(that)
+
+  final def catchSome(
```

Comment:
This doesn't have the idiomatic signature of `catchSome`, so I'd recommend a new name. Also does it really make sense to have a single fallback? I can create a generic signature here if you can explain to me the purpose of the method better.

Reply:
That makes sense, I can change the name of the method. The purpose of this method is to allow users to build their own "fallback" methods if required. For example, a user may define a new config source and may want finer control when falling back on another config source. This method is probably not required so for simplicity, maybe it is best to remove this method.